packages/timeline-state-resolver/src/integrations/tricaster/index.ts

```diff
@@ -0,0 +1,176 @@
+		})
+		this._connection.on('disconnected', (_reason) => {
+			this._setConnected(false)
+		})
+		this._connection.on('error', (reason) => {
+			this.emit('error', 'TriCasterConnection', reason)
+		})
+		this._connection.connect()
+		return initPromise
+	}
+
+	private _setInitialState(shortcutStateXml: string): void {
+		if (!this._stateDiffer) {
+			throw new Error('State Differ not available')
+		}
+		const time = this.getCurrentTime()
+		const state = this._stateDiffer.shortcutStateConverter.getTriCasterStateFromShortcutState(shortcutStateXml)
+		this.setState(state, time)
+	}
+
+	private _connectionChanged(): void {
+		this.emit('connectionChanged', this.getStatus())
+	}
+
+	private _setConnected(connected: boolean): void {
+		if (this._connected !== connected) {
+			this._connected = connected
+			this._connectionChanged()
+		}
+	}
+
+	/** Called by the Conductor a bit before handleState is called */
+	prepareForHandleState(newStateTime: number): void {
+		// clear any queued commands later than this time:
+		this._doOnTime.clearQueueNowAndAfter(newStateTime)
+		this.cleanUpStates(0, newStateTime)
+	}
+
+	handleState(newState: TimelineState, newMappings: Mappings): void {
+		super.onHandleState(newState, newMappings)
+		if (!this._initialized || !this._stateDiffer) {
+			// before it's initialized don't do anything
+			this.emit('warning', 'TriCaster not initialized yet')
+			return
+		}
+
+		const previousStateTime = Math.max(this.getCurrentTime(), newState.time)
+		const oldState = this.getStateBefore(previousStateTime)?.state ?? this._stateDiffer.getDefaultState()
+
+		const newTriCasterState = this._stateDiffer.timelineStateConverter.getTriCasterStateFromTimelineState(
+			newState,
+			newMappings,
+			this.deviceId
+		)
+
+		const commandsToAchieveState = this._stateDiffer.getCommandsToAchieveState(newTriCasterState, oldState)
+
+		// clear any queued commands later than this time:
+		this._doOnTime.clearQueueNowAndAfter(previousStateTime)
+
+		// add the new commands to the queue:
+		this._addToQueue(commandsToAchieveState, newState.time)
+
+		// store the new state, for later use:
+		this.setState(newTriCasterState, newState.time)
+	}
+
+	clearFuture(clearAfterTime: number): void {
+		// Clear any scheduled commands after this time
+		this._doOnTime.clearQueueAfter(clearAfterTime)
+	}
+
+	async terminate(): Promise<boolean> {
+		this._doOnTime.dispose()
+		this._connection?.close()
+		return Promise.resolve(true)
+	}
+
+	getStatus(): DeviceStatus {
+		let statusCode = StatusCode.GOOD
+		const messages: Array<string> = []
+
+		if (!this._connected) {
+			statusCode = StatusCode.BAD
+			messages.push('Not connected')
+		}
+
+		return {
+			statusCode: statusCode,
+			messages: messages,
+			active: this.isActive,
+		}
+	}
+
+	async makeReady(okToDestroyStuff?: boolean): Promise<void> {
+		if (okToDestroyStuff) {
+			// do something?
+		}
+	}
+
+	get canConnect(): boolean {
+		return false
```

Comment:
Is this suppose to return a static `false`? 
If a `TricasterDevice` is asked, if it can connect, should it always say no?

Might be me who has not grasped how Tricaster functions yet, so please enlighten me.

Reply:
It should return `true` instead, because the method seems to really be answering the question "is this device integration capable of establishing a persistent connection, status of which will be reported after it's established?"